components/textfield/index.css

```diff
@@ -12,56 +12,74 @@
+	--spectrum-textfield-width: var(--spectrum-field-default-width-medium);
 	--spectrum-textfield-min-width: var(--spectrum-text-field-minimum-width-multiplier);
-	--spectrum-textfield-corner-radius: var(--spectrum-corner-radius-100);
-
-	--spectrum-textfield-border-color: var(--spectrum-gray-500);
-	--spectrum-textfield-border-color-hover: var(--spectrum-gray-600);
-	--spectrum-textfield-border-color-focus: var(--spectrum-gray-800);
-	--spectrum-textfield-border-color-focus-hover: var(--spectrum-gray-900);
-	--spectrum-textfield-border-color-keyboard-focus: var(--spectrum-gray-900);
-
-	--spectrum-textfield-border-width: var(--spectrum-border-width-100);
+	--spectrum-textfield-border-width: var(--spectrum-border-width-200);
+	--spectrum-textfield-corner-radius: var(--spectrum-corner-radius-medium-size-medium);
 
 	/* default spacing */
-	--spectrum-textfield-spacing-inline-quiet: var(--spectrum-field-edge-to-text-quiet);
+	--spectrum-textfield-spacing-inline: var(--spectrum-component-edge-to-text-100);
 	--spectrum-textfield-spacing-block-start: var(--spectrum-component-top-to-text-100);
 	--spectrum-textfield-spacing-block-end: var(--spectrum-component-bottom-to-text-100);
-	--spectrum-textfield-spacing-block-quiet: var(--spectrum-field-edge-to-border-quiet);
 
-	/* default label spacing */
+	/* default field label and help text spacing */
 	--spectrum-textfield-label-spacing-block: var(--spectrum-field-label-to-component);
-
-	/* default help text spacing */
+	--spectrum-textfield-label-spacing-inline-side-label: var(--spectrum-spacing-200);
 	--spectrum-textfield-helptext-spacing-block: var(--spectrum-help-text-to-component);
 
+	/* default icon size */
+	--spectrum-textfield-icon-size-invalid: var(--spectrum-workflow-icon-size-100);
+	--spectrum-textfield-icon-size-valid: var(--spectrum-checkmark-icon-size-100);
+
 	/* default icon spacing - invalid */
-	--spectrum-textfield-icon-spacing-inline-end-quiet-invalid: var(--spectrum-field-edge-to-alert-icon-quiet);
+	--spectrum-textfield-icon-spacing-inline-start-invalid: var(--spectrum-text-to-visual-100);
+	--spectrum-textfield-icon-spacing-inline-end-invalid: var(--spectrum-field-edge-to-alert-icon-medium);
+	--spectrum-textfield-icon-spacing-block-invalid: var(--spectrum-component-top-to-workflow-icon-100);
 
 	/* default icon spacing - valid */
-	--spectrum-textfield-icon-spacing-inline-end-quiet-valid: var(--spectrum-field-edge-to-validation-icon-quiet);
+	--spectrum-textfield-icon-spacing-inline-start-valid: var(--spectrum-text-to-visual-100);
+	--spectrum-textfield-icon-spacing-inline-end-valid: var(--spectrum-field-edge-to-validation-icon-medium);
+	--spectrum-textfield-icon-spacing-block-valid: var(--spectrum-field-top-to-validation-icon-medium);
 
 	/* font styles */
 	--spectrum-textfield-font-family: var(--spectrum-sans-font-family-stack);
 	--spectrum-textfield-font-weight: var(--spectrum-regular-font-weight);
+	--spectrum-textfield-font-style: var(--spectrum-default-font-style);
```

Comment:
You'll see that `font-style` was added in multiple places because it wasn't here before. I'm not sure if there's an argument for keeping it out of the CSS since it wasn't here before, but since it was in the spec, I added it.

I would also love to hear any feedback on whether we really need to be offering mods for all of these and whether the input and character count should really have different custom properties (is it useful to have `--spectrum-textfield-font-family` and also `--spectrum-textfield-character-count-font-family` for instance). We set fonts on the character count, placeholder, and input, and that all feels a little heavy-handed.